### PR TITLE
Issue-436:Convert Pytest-CVP Login Banner test case to Vane #436

### PR DESCRIPTION
**Pull request by paragpande24**
# Please include a summary of the changes
CDescribe the solution you'd like
PyTest-CVP Login Banner test case should have the following:

Use Vane test case styling
Use Master definition file to simplify input
Add test case steps to the test case
Test should be added to top-level directory nrfu_tests
Use markers nrfu_test, security
Use Test ID: NRFU5.5

# Any specific logic/part of code you need extra attention on
 From the output of  'show banner login' and verifying that the login banner is correct.

# Include the Issue number and link
#436 

# List/Attach any dependencies/past issues that are required for this change/provide context

# Type of change

- - [ ] Bug fix (non-breaking change which fixes an issue)
- - [ ] New feature (non-breaking change which adds functionality)
- - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- - [ ] This change requires a documentation update
- - [x] Other (NRFU Testcase)

# Effort required on reviewers end

- - [x] Easy
- - [ ] Medium
- - [ ] Hard 

# How Has This Been Tested?
Unit Tests:

- Tested for all the verifying that the login banner is correct.: Passed

[PASS-436-NEW.docx](https://github.com/aristanetworks/vane/files/12519095/PASS-436-NEW.docx)

- Tested if login banner is not enabled on switch: Failed.
 
[FAIL-436-NEW.docx](https://github.com/aristanetworks/vane/files/12519093/FAIL-436-NEW.docx)

- Tested  for verifying that the login banner is correct: Passed (With generate test definition True and J2 True )
   
[PASS-436-NEW-WITH-J2.docx](https://github.com/aristanetworks/vane/files/12519092/PASS-436-NEW-WITH-J2.docx)

HTML reports:

[HTML-436-NEW.zip](https://github.com/aristanetworks/vane/files/12519091/HTML-436-NEW.zip)

-Pass report after adding logging and f-string changes.

[436-PASS-AFTER-LOGGING.docx](https://github.com/aristanetworks/vane/files/12606311/436-PASS-AFTER-LOGGING.docx)

[436-PASS-AFTER-LOGGING.zip](https://github.com/aristanetworks/vane/files/12606321/436-PASS-AFTER-LOGGING.zip)

## Bug fix

    Include before and after snapshots/screenshots/changed logs
    
## New feature

    Ensure current test cases pass and include newer test cases if required
    
# CI pipeline result

- - [ ] Pass
- - [ ] Fail
  
  If it fails, explain the reason and whether or not we should ignore the failure
  
# Verify Documentation Update

    If applicable, ensure documentation has been updated for ReadMe, Getting Started guide, and Style guide
    
# Additional comments

### CODE REVIEW

File: nrfu_tests/test_security_rp_login_banner.py [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        tops = tests_tools.TestOps(tests_definitions, TEST_SUITE, dut)
+        self.output = ""
+        tops.actual_output = {"login_banner_found": False}
+
+        # Forming output message if test result is passed
+        tops.output_msg = "Expected login banner is present on the switch."
+
+        try:
+            """
+            TS: Running `show banner login` command on a DUT and verifying that the
```
Comment (hsisalgslab) [inline nrfu_tests/test_security_rp_login_banner.py:45]:
```suggestion
            TS: Running `show banner login` command on device and verifying that the expected 

```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            login_banner = output.get("loginBanner")
+
+            if login_banner:
+                tops.actual_output = {"login_banner_found": True}
+
+            # forming output message if test result is fail
+            if tops.expected_output != tops.actual_output:
+                if not tops.actual_output["login_banner_found"]:
+                    tops.output_msg = f"Login banner is not configured on {tops.dut_name}"
```
Comment (hsisalgslab) [inline nrfu_tests/test_security_rp_login_banner.py:64]:
```suggestion
                    tops.output_msg = f" Expected login banner is not configured on {tops.dut_name}"

```

File: nrfu_tests/test_definition.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_security_rp_login_banner
+    description: Testcase for verification of login banner is configured.
```
Comment (hsisalgslab) [inline nrfu_tests/test_definition.yaml:4]:
```suggestion
    description: Testcase for verification of security rp login banner
```
pls update on all applicable cases.

File: nrfu_tests/master_def.yaml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+global_dut_filter: # This stanza is optional. If it is not provided by users than all test cases should be run on each dut.
```
Comment (hsisalgslab) [inline nrfu_tests/master_def.yaml:1]: as we discussed in DSU, lets not add master def for now.

**Review by hsisalgslab — APPROVED** [review]

**Review by hsisalgslab — APPROVED** [review]

File: nrfu_tests/test_definition.yaml [hunk 5]
```diff
@@ -0,0 +1,13 @@
+- name: nrfu_tests
+  testcases:
+  - name: test_security_rp_login_banner
+    description: Testcase for verification of security rp login banner.
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:4]:
```suggestion
    description: Testcase for verification of login banner.
```

File: nrfu_tests/test_definition.yaml [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+    test_id: NRFU5.5
+    show_cmd: show banner login
+    expected_output:
+      login_banner_found : True
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:8]:
```suggestion
      login_banner_found: True
```

File: nrfu_tests/test_definition.yaml [hunk 7]
```diff
@@ -0,0 +1,13 @@
...
+    test_criteria: Expected login banner should be present on the switch. # Setting test case’s pass / fail criteria for reporting
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_definition.yaml:9]:
```suggestion
    test_criteria: Login banner should be found on the device. # Setting test case’s pass / fail criteria for reporting
```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 8]
```diff
@@ -0,0 +1,79 @@
...
+from vane.logger import logger
+from vane.config import dut_objs, test_defs
+from vane import tests_tools
+
+TEST_SUITE = "nrfu_tests"
+
+
+@pytest.mark.nrfu_test
+@pytest.mark.security
+class SecurityRpLoginBannerTests:
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:19]:
```suggestion
class LoginBannerTests:

```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+        tops.actual_output = {"login_banner_found": False}
+
+        # Forming output message if test result is passed
+        tops.output_msg = "Expected login banner is present on the switch."
+
+        try:
+            """
+            TS: Running `show banner login` command on device and verifying that the expected
+            login banner is correct.
+            """
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:47]:
```suggestion
            TS: Running `show banner login` command on device and verifying that the login banner is found on the device.
            """

```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 10]
```diff
@@ -0,0 +1,79 @@
...
+                "On device %s, output of %s command is: \n%s\n",
+                tops.dut_name,
+                tops.show_cmd,
+                output,
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            login_banner = output.get("loginBanner")
+
+            if login_banner:
+                tops.actual_output = {"login_banner_found": True}
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:59]:
```suggestion
tops.actual_output = {"login_banner_found": bool( login_banner)}

```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 11]
```diff
@@ -0,0 +1,79 @@
...
+
+            # forming output message if test result is fail
+            if tops.expected_output != tops.actual_output:
+                if not tops.actual_output["login_banner_found"]:
+                    tops.output_msg = (
+                        f"\nExpected login banner is not configured on {tops.dut_name}."
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:65]:
```suggestion
                        f"\nLogin banner is not found on the device"

```

File: nrfu_tests/test_security_rp_login_banner.py [hunk 12]
```diff
@@ -0,0 +1,79 @@
...
+
+        # Forming output message if test result is passed
+        tops.output_msg = "Expected login banner is present on the switch."
+
+        try:
+            """
+            TS: Running `show banner login` command on device and verifying that the expected
+            login banner is correct.
+            """
+            output = dut["output"][tops.show_cmd]["json"]
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:48]: Why command output is not added in docx report?

image

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

File: nrfu_tests/test_security_rp_login_banner.py [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+                output,
+            )
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            login_banner = output.get("loginBanner")
+            tops.actual_output = {"login_banner_found": bool(login_banner)}
+
+            # forming output message if test result is fail
+            if tops.expected_output != tops.actual_output:
+                if not tops.actual_output["login_banner_found"]:
+                    tops.output_msg = "\nLogin banner is not found on the device."
```
Comment (vishwajeetgslab) [inline nrfu_tests/test_security_rp_login_banner.py:62]:
```suggestion
                    tops.output_msg = "Login banner is not found on the device."

```

**Review by vishwajeetgslab — CHANGES_REQUESTED** [review]

**Review by vishwajeetgslab — APPROVED** [review]

File: nrfu_tests/test_security_rp_login_banner.py [hunk 14]
```diff
@@ -0,0 +1,75 @@
...
+            self.output += f"\nOutput of {tops.show_cmd} command is: \n{output}"
+            login_banner = output.get("loginBanner")
+            tops.actual_output = {"login_banner_found": bool(login_banner)}
+
+            # forming output message if test result is fail
+            if tops.expected_output != tops.actual_output:
+                if not tops.actual_output["login_banner_found"]:
+                    tops.output_msg = "Login banner is not found on the device."
+
+        except (AssertionError, AttributeError, LookupError, EapiError) as excep:
```
Comment (MaheshGSLAB) [inline nrfu_tests/test_security_rp_login_banner.py:64]:
```suggestion
        except (AttributeError, LookupError, EapiError) as excep:

```

**Review by kgrozis-arista — APPROVED** [review]

**Pull request merged**